tests/test_examples.py

```
@@ -0,0 +1,386 @@
+    them as separate arguments. We don't care about multiple spaces collapsing into
+    one, since the point is to just have spaces in the installation path -- one
+    would be enough too :)
+    This is why we have this weird .split() thingy down there in `/D=...`.
+    """
+    if not sys.platform.startswith("win"):
+        raise ValueError("Can only run .exe installers on Windows")
+    if "NSIS_USING_LOG_BUILD" not in os.environ:
+        warnings.warn(
+            "Windows installers are tested with NSIS in silent mode, which does "
+            "not report errors on exit. You should use logging-enabled NSIS builds "
+            "to generate an 'install.log' file this script will search for errors "
+            "after completion."
+        )
+    cmd = ["cmd.exe", "/c", "start", "/wait", installer, "/S", *f"/D={install_dir}".split()]
+    _execute(cmd)
+
+    # Windows installers won't raise exit codes so we need to check the log file
+    error_lines = []
+    try:
+        log_is_empty = True
+        with open(os.path.join(install_dir, "install.log"), encoding="utf-16-le") as f:
+            for line in f:
+                log_is_empty = False
+                if ":error:" in line:
+                    error_lines.append(line)
+        if log_is_empty:
+            error_lines.append("Logfile was unexpectedly empty!")
+    except Exception as exc:
+        error_lines.append(
+            f"Could not read logs! {exc.__class__.__name__}: {exc}\n"
+            "This usually means that the destination folder could not be created.\n"
+            "Possible causes: permissions, non-supported characters, long paths...\n"
+            "Consider setting 'check_path_spaces' and 'check_path_length' to 'False'."
+        )
+    if error_lines:
+        raise AssertionError("\n".join(error_lines))
+
+
+def _run_uninstaller_exe(install_dir):
+    # Now test the uninstallers
+    if " " in str(install_dir):
```

Better to raise an exception here? Send safer than silently allowing the test to pass (assuming that's the outcome of short circuiting here)

---

That's a known issue we couldn't fix in CI (works locally), but I agree we could at least add a warning!